From 8b5a69e0deda506dc7da18622922b1736c3bc6a4 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Fri, 15 Aug 2014 19:43:56 +0100
Subject: lircd: utilise lirc_log_reopen

-remove code duplication
---
 daemons/lircd.c | 23 ++++-------------------
 1 file changed, 4 insertions(+), 19 deletions(-)

diff --git a/daemons/lircd.c b/daemons/lircd.c
index 97cc70b..fb9d37a 100644
--- a/daemons/lircd.c
+++ b/daemons/lircd.c
@@ -420,26 +420,11 @@ void dosighup(int sig)
 	struct stat s;
 	int i;
 
-	/* reopen logfile first */
+	/* can we print error messagees */
 
-	if (! lirc_log_use_syslog()){
-		logprintf(LOG_INFO, "closing logfile");
-		if (-1 == fstat(fileno(lf), &s)) {
-			dosigterm(SIGTERM);	/* shouldn't ever happen */
-		}
-  		lirc_log_close();
-                lirc_log_open("lircd", nodaemon, debug_opt);
-		lf = fopen(logfile, "a");
-		if (lf == NULL) {
-			/* can't print any error messagees */
-			dosigterm(SIGTERM);
-		}
-		logprintf(LOG_INFO, "reopened logfile");
-		if (-1 == fchmod(fileno(lf), s.st_mode)) {
-			logprintf(LOG_WARNING, "could not set file permissions");
-			logperror(LOG_WARNING, NULL);
-		}
-	}
+	/* reopen log */
+	if (lirc_log_reopen() != 0)
+		dosigterm(SIGTERM); /* shouldn't ever happen */
 
 	config();
 
-- 
1.8.4

