From 7e0cfce60359a8831a777951b76bfb1559d5a732 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Sun, 1 Oct 2017 11:08:43 +0100
Subject: embeddedartwidgetbar, add removal of embedded art support

---
 .../quodlibet/ext/covers/embeddedart_widgetbar.py  | 36 ++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/quodlibet/quodlibet/ext/covers/embeddedart_widgetbar.py b/quodlibet/quodlibet/ext/covers/embeddedart_widgetbar.py
index abe6b8aa1..101a73c96 100644
--- a/quodlibet/quodlibet/ext/covers/embeddedart_widgetbar.py
+++ b/quodlibet/quodlibet/ext/covers/embeddedart_widgetbar.py
@@ -390,6 +390,36 @@ class EmbeddedArtBox(Gtk.HBox):
             except AudioFileError:
                 print_exc()
 
+    def _remove_image(self):
+
+        for w in self.get_selected_image_widgets():
+            if not w.song.can_change_images:
+                ext = os.path.splitext(w.song['~filename'])[1][1:]
+                print_d("skipping unsupported song type %r [%s]"
+                        % (ext, w.song['~filename']))
+                continue
+
+            images = w.song.get_images()
+            if len(images) == 1:
+                try:
+                    w.song.clear_images()
+                    self._remove_widget_by_image_widget(w)
+                except AudioFileError:
+                    print_exc()
+            else:
+                # iterate and compare to find this image
+
+                fo = open(w.image.name, 'rb')
+                for image in images:
+                    if self.__file_equals_embeddedimage(fo, image):
+                        try:
+                            if not w.song.remove_image(image):
+                                print_d("failed to remove image for song %r"
+                                        % w.song)
+                        except AudioFileError:
+                            print_exc()
+                        break
+
     def _set_image(self):
 
         for w in self.get_selected_image_widgets():
@@ -518,6 +548,12 @@ class EmbeddedArtWidgetBarPlugin(UserInterfacePlugin, EventPlugin):
             lambda *_: self.__embeddedart_box._clear_images())
         self.__controls_box.pack_start(clear_button, False, False, 5)
 
+        remove_button = Gtk.Button(_(u"Remove"))
+        remove_button.connect(
+            "button-press-event",
+            lambda *_: self.__embeddedart_box._remove_image())
+        self.__controls_box.pack_start(remove_button, False, False, 5)
+
         self.__select_count = 0
         self.__total_count = 0
         self.__update_count()
-- 
2.14.1

