From f7dc9be1cf8daaf28720db3ff6876c4eebca9b4a Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 6 Sep 2017 17:13:28 +0100
Subject: remove multiline debug statement

---
 quodlibet/quodlibet/qltk/quodlibetwindow.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/quodlibet/quodlibet/qltk/quodlibetwindow.py b/quodlibet/quodlibet/qltk/quodlibetwindow.py
index 45b2d28..2d16671 100644
--- a/quodlibet/quodlibet/qltk/quodlibetwindow.py
+++ b/quodlibet/quodlibet/qltk/quodlibetwindow.py
@@ -940,9 +940,11 @@ class QuodLibetWindow(Window, PersistentWindowMixin, AppWindow):
                 [val for key, val in self.ui_elements.items()
                     if key in extra])
 
-        print_d("%d ui elements live:" % len(self.ui_elements_ordered))
-        for w in self.ui_elements_ordered:
-            print_d("%r" % w)
+        print_d("%d ui elements live:\n%s" %
+                    (len(self.ui_elements_ordered),
+                     '\n'.join(list(
+                         map(lambda s: s[0] + ' | ' + str(s[1]),
+                             list(zip(order, self.ui_elements_ordered)))))))
 
         return self.ui_elements_ordered
 
-- 
1.8.4

